app/src/main/webapp/js/services.js

```diff
@@ -42,51 +42,84 @@ esbMessageAdminServices.service('EsbMessageService', ['$http', '$q', 'Globals',
     };
 
     self.getSyncKeysTree = function() {
-        return $http({method: 'GET', url: "api/key/tree/Entities"});
+        return self.getTree("Entities");
     };
 
     self.getSearchKeysTree = function() {
-        return $http({method: 'GET', url: "api/key/tree/SearchKeys"});
+        return self.getTree("SearchKeys");
     };
 
-    self.addKey = function(argParentId, argName, argType, argValue) {
+    self.respondSuccess = function(argResponse) {
+        if(argResponse.status==="Success") {
+            return argResponse.tree;
+        } else {
+            messageCenterService.add('danger', argResponse.errorMessage, {status: messageCenterService.status.permanent});
+            return null;
+        }
+    }
+
+    self.respondError= function()  {
```

Comment:
Maybe different terminology than `respond` would be nice for these methods (`respondSuccess` and `respondError`). In this context "response" / "respond" is used when getting a response from a remote call, and these are not really responding to anything remotely. It's purely a style thing, but, just a suggestion. Maybe `alertSuccess` / `alertError` ... this is what bootstrap calls that sort of thing, at least: http://getbootstrap.com/components/#alerts